### ISSUE TITLE
accounts:balance `noteToNullifier` error

### ISSUE THREAD

**Issue by wd021** [iron-fish/ironfish#1050]
### What happened?

accounts:balance broken. see logs below 👇 

### Version

v0.1.24

### Relevant log output

```shell
root@ip-172-31-22-211:~/ironfish/ironfish-cli# yarn start:once accounts:balance
yarn run v1.22.17
$ yarn build && yarn start:js accounts:balance
$ tsc -b
$ cross-env OCLIF_TS_NODE=0 IRONFISH_DEBUG=1 node bin/run accounts:balance
All decryptable notes should be in the noteToNullifier map
Done in 1.63s.
root@ip-172-31-22
```

**Comment by Kupuyc** [comment]
Seems like it depends of instance. Could the database have been corrupted due to the "Could not expand shared link" error?

```
root@nfd8a11:~# ironfish accounts:balance
yarn run v1.22.17
$ yarn build && yarn start:js accounts:balance
$ tsc -b
$ cross-env OCLIF_TS_NODE=0 IRONFISH_DEBUG=1 node bin/run accounts:balance
The account balance is:    $IRON 0.00000000 ($ORE 0)
Amount available to spend: $IRON 0.00000000 ($ORE 0)
Done in 1.26s.
```

```
root@srv52423:~# ironfish accounts:balance
yarn run v1.22.17
$ yarn build && yarn start:js accounts:balance
$ tsc -b
$ cross-env OCLIF_TS_NODE=0 IRONFISH_DEBUG=1 node bin/run accounts:balance
The account balance is:    $IRON 2,120.02000069 ($ORE 212,002,000,069)
Amount available to spend: $IRON 2,120.02000069 ($ORE 212,002,000,069)
Done in 2.18s.
```

**Comment by wd021** [comment]
@Kupuyc what's the "Could not expand shared link" error? not sure I've run into that over here.

**Comment by Kupuyc** [comment]
It begun in 0.1.22. You may see it in #1022 issue. Probably, in case that issue is consistently repeated on your database, it will be good way to debug it.

**Comment by terrancewong** [comment]
0.1.24 also suffers.

```
./ironfish-cli/bin/ironfish accounts:balance                                                        
All decryptable notes should be in the noteToNullifier map
```

node launched with `--rpc.tcp.secure`. And it worked yesterday.

**Comment by terrancewong** [comment]
Thanks, but `accounts:rescan` is almost as slow as resync entire chain....

**Comment by wd021** [comment]
@terrancewong are you on the latest version?

**Comment by terrancewong** [comment]
0.1.24, I'll try 0.1.25 now

**Comment by terrancewong** [comment]
## version 

0.1.25

## command line

```
 % ./ironfish-cli/bin/ironfish accounts:rescan 
Rescanning Transactions... Scanning Complete
 % ./ironfish-cli/bin/ironfish accounts:balance
All decryptable notes should be in the noteToNullifier map
```

**Comment by terrancewong** [comment]
`reset` and started another node, `accounts:balance` worked.

**Comment by NullSoldier** [comment]
This is temporarily fixable by running `accounts:rescan --reset` but were in the process of really fixing this with wallet 2.0.

**Comment by hughy** [comment]
Errors like this one should be resolved by the Wallet 2.0 refactor released in [v0.1.48](https://github.com/iron-fish/ironfish/releases/tag/v0.1.48)

Please open a new issue if you run into this again!
